### PR TITLE
app-misc/flirc: Changed to new package source, including desktop files and udev rules

### PR DESCRIPTION
**Pull request by github-k8n**
Closes: https://bugs.gentoo.org/668596

### CODE REVIEW

File: app-misc/flirc/Manifest [hunk 1]
```diff
@@ -1,2 +1,2 @@
-DIST flirc-3.11.6_amd64.deb 8994136 BLAKE2B 0dfcf230daee195a844aaf746e8f1cd259eb9b3eb7203483040c3b0c7ef2d7eded09064ed6dd30e3f72568f59dbddf3d96c5b26a84cad5f9e2e464c7e114e4bf SHA512 06dfb22653dac1efe3e67eab8b2806a4760f4b3646a712222658d48cfb3defe3abb9f5fe8a0bfbb8dfc332a0f4e89691c304402d3d5ea28328bd5b6619069ece
```
Comment (mgorny) [inline app-misc/flirc/Manifest]: Please split addition and removal into separate commits.

File: app-misc/flirc/flirc-3.22.2.ebuild [hunk 2]
```diff
@@ -3,17 +3,17 @@
...
 EAPI=6
 
-inherit desktop udev unpacker
+inherit desktop udev
 
 DESCRIPTION="Allows you to pair any remote control with your computer or media center"
 HOMEPAGE="https://flirc.tv/"
 SRC_URI="
-	amd64?	( https://packagecloud.io/Flirc/repo/packages/ubuntu/artful/flirc_${PV}_amd64.deb/download.deb  -> ${P}_amd64.deb )
-	arm?	( https://packagecloud.io/Flirc/repo/packages/debian/stretch/flirc_${PV}_armhf.deb/download.deb -> ${P}_armhf.deb )"
```
Comment (mgorny) [inline app-misc/flirc/flirc-3.22.2.ebuild]: I see upstream has arm archive as well, so why remove it?

File: app-misc/flirc/flirc-3.22.2.ebuild [hunk 3]
```diff
@@ -3,17 +3,17 @@
...
+	amd64?	( http://apt.flirc.tv/arch/x86_64/flirc.latest.x86_64.tar.gz -> ${P}_amd64.tar.gz )
```
Comment (mgorny) [inline app-misc/flirc/flirc-3.22.2.ebuild]: Any chance you could find versioned URLs somewhere? This is going to suck hard given it's mirror-restricted.

File: app-misc/flirc/flirc-3.22.2.ebuild [hunk 4]
```diff
@@ -3,17 +3,17 @@
 
 EAPI=6
```
Comment (mgorny) [inline app-misc/flirc/flirc-3.22.2.ebuild]: EAPI 7 should be possible here.

File: app-misc/flirc/flirc-3.22.2.ebuild [hunk 5]
```diff
@@ -35,13 +35,11 @@ RDEPEND="virtual/libusb:1
 QA_PREBUILT="/usr/bin/*"
 
 src_install () {
-	udev_newrules etc/udev/rules.d/99-flirc.rules 51-flirc.rules
```
Comment (mgorny) [inline app-misc/flirc/flirc-3.22.2.ebuild]: I'm confused. Doesn't upstream provide them anymore? FWICS that `apt.flirc.tv` thing is suited for Debian, so they should have complete Debian packages somewhere. Maybe figure out how apt gets data from it, so you could find them.

File: app-misc/flirc/flirc-3.22.2.ebuild [hunk 6]
```diff
@@ -35,13 +35,11 @@ RDEPEND="virtual/libusb:1
...
-	dobin usr/bin/flirc_util
+	udev_newrules "${FILESDIR}"/51-flirc.rules 51-flirc.rules
+	dobin "Flirc-${PV}"/flirc_util
 	if use qt5 ; then
-		doman usr/share/doc/flirc/Flirc.1
-		dobin usr/bin/Flirc
-		doicon usr/share/pixmaps/Flirc.png
-		domenu usr/share/applications/Flirc.desktop
+		dobin "Flirc-${PV}"/Flirc
+		doicon "${FILESDIR}"/Flirc.png
```
Comment (mgorny) [inline app-misc/flirc/flirc-3.22.2.ebuild]: Binary files in FILESDIR are forbidden.

**Comment by github-k8n** [comment]
Ok, hope that works now, changed to versioned .deb source, updated to 3.22.4, updated eapi to 7 and split deletion and addition of new files (though the manifest modification is in the "new" files, not sure if that should be "removed" and "added"?, let me know if that is an issue, then i can rebase/split it again)

**Comment by mgorny** [comment]
Thanks, that's nicer. Could you also adjust commit messages to match the Gentoo standard? GLEP 66 is your friend.
